source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -329,14 +396,28 @@ class Filter : public Http::StreamFilter, Logger::Loggable<Logger::Id::lua> {
   // Http::StreamDecoderFilter
   Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap& headers,
                                           bool end_stream) override {
-    return doHeaders(request_stream_wrapper_, request_coroutine_, decoder_callbacks_,
-                     config_->requestFunctionRef(), headers, end_stream);
+    std::string name = GLOBAL_SCRIPT_NAME;
+    const auto config_per_route = getConfigPerRoute(decoder_callbacks_.callbacks_);
+    if (config_per_route) {
+      // Filter is disabled by route config
+      if (config_per_route->disabled()) {
+        return Http::FilterHeadersStatus::Continue;
+      }
+      name = config_per_route->name().empty() ? name : config_per_route->name();
+      script_config_ = config_per_route->scriptConfig();
+    }
+    script_config_ = script_config_ ? script_config_ : config_->scriptConfig(name);
+    if (!script_config_) {
+      return Http::FilterHeadersStatus::Continue;
+    }
```

Comment:
Can we wrap this as a helper?

Reply:
If we have per filter config with defined inline code (for example from RDS), we should not bother to check on the "name" (as a lookup key to the named Lua codes)?